hooks/checkov.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+#######################################################################
+# Wrapper function for hook. Determines run mode based on bool set by
+# argparse, and triggers that run mode.
+# Globals:
+#   None
+# Arguments:
+#   None
+# Returns:
+#   0 if successful, non-zero on error
+#######################################################################
+function main {
+  parse_arguments "$@"
+  # shellcheck disable=SC2153 # False positive
+  if [[ ${CHANGE_DIRECTORY_SCAN} == true ]]; then
+    # directories containing changed .tf files only
+    directory_runner
+  elif [[ ${CHANGE_FILE_SCAN} == true ]]; then
+    # changed .tf files only
+    file_runner
+  else
+    # whole repository
+    checkov "${ARGS[@]}" -d .
+  fi
+}
+
+#######################################################################
+# Parses arguments provided to the hook to filter out changed files and
+# args for the hook, set run mode (if present) and remove run mode toggles
+# from the args as they're not valid for checkov. Enables backwards
+# compatibility for those using checkov with no args, or without --args
+# Globals:
+#   None
+# Arguments:
+#   None
+# Returns:
+#   0 if successful, non-zero on error
+#   Creates global arrays ARGS, FILES, may create a run mode global
+#######################################################################
+function parse_arguments {
+  declare -a -g ARGS=()
+  declare -a -g FILES=()
+
+  argv=("$@")
+  pattern='^.*\.tf$'
```

Comment:
The any_chars_from_beginning part of pattern makes no much sense and can be dropped
```suggestion
  pattern='\.tf$'
```